### DISCUSSION TITLE
Load custom font on C#

### DISCUSSION THREAD

**Discussion by rogerioferro** [ImageMagick/ImageMagick#7545] [Help]
### ImageMagick version

13.10.0

### Operating system, version and so on

Windows / Linux

### Description

I can't load custom font in c#.
I have a ttf file at a byte array in memory and I want use this font on image magick.
I am using this package: https://www.nuget.org/packages/Magick.NET-Q8-AnyCPU

**Comment by dlemstra** [comment]
You are not providing a lot of information. If you want to use a font you will need to save it to disk first and then use it as a font file in the code where you need this font.

**Reply by rogerioferro** [reply]
My code are running in a docker image. I don't have access to the disk. All resource are accessed in memory.

**Reply by rogerioferro** [reply]
I read resources in C# with a code like this:
private static byte[] GetFontResourceBytes(Assembly assembly, string fontResourceName)
{
    var resourceStream = assembly.GetManifestResourceStream(fontResourceName);
    if (resourceStream == null)
        throw new Exception(string.Format("Unable to find font '{0}' in embedded resources.", fontResourceName));
    var fontBytes = new byte[resourceStream.Length];
    resourceStream.Read(fontBytes, 0, (int)resourceStream.Length);
    resourceStream.Close();
    return fontBytes;
}

**Reply by dlemstra** [reply]
You don't have access to `/tmp`? And as I said before a font needs a file on disk.

**Reply by rogerioferro** [reply]
Save on /tmp worked.

Thanks.
